models/classes/Security/AccessTokenRequestValidator.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\models\classes\Security;
+
+use OAT\Library\Lti1p3Core\Registration\RegistrationRepositoryInterface;
+use OAT\Library\Lti1p3Core\Service\Server\Validator\AccessTokenRequestValidator as Lti1p3AccessTokenRequestValidator;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoLti\models\classes\LtiProvider\InvalidLtiProviderException;
+use oat\taoLti\models\classes\LtiProvider\LtiProvider;
+use oat\taoLti\models\classes\LtiProvider\LtiProviderService;
+use oat\taoLti\models\classes\Platform\Repository\Lti1p3RegistrationRepository;
+use oat\taoLti\models\classes\Platform\Service\AccessTokenRequestValidatorInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use tao_models_classes_UserException;
+
+class AccessTokenRequestValidator extends ConfigurableService implements AccessTokenRequestValidatorInterface
+{
+    /** @var Lti1p3AccessTokenRequestValidator */
+    private $validator;
+
+    /** @var LtiProvider */
+    private $ltiProvider;
+
+    /** @var string */
+    private $role;
+
+    public function withLtiProvider(LtiProvider $ltiProvider): AccessTokenRequestValidatorInterface
+    {
+        $this->ltiProvider = $ltiProvider;
+
+        return $this;
+    }
+
+    public function withRole(string $role): AccessTokenRequestValidatorInterface
+    {
+        $this->role = $role;
+
+        return $this;
+    }
+
+    public function validate(ServerRequestInterface $request): void
```

Comment:
i think we need to handle the the `MissingScopeException` and also `tao_models_classes_UserException` either inside the interface description either here

Reply:
Added missing `@throws` 
https://github.com/oat-sa/extension-tao-lti/pull/268/commits/e9b3da489c07f08e1140cf7404afa74dd08f2fad